### DISCUSSION TITLE
Need information on enabling AES-GCM ciphers in openSSL 1.0.2u

### DISCUSSION THREAD

**Discussion by sireeshkumartholeti** [openssl/openssl#21483] [Q&A] [answered]
Hi,
In our project we are using ACE (6.3.3) and openSSL (1.0.2u), we are trying to make AES-GCM ciphers as default ciphers in our project.
However, we are not able to find any reference of enabling AES-GCM ciphers, Can anyone please help on how to make AES-GCM ciphers as default.

**Comment by t8m** [comment, marked as answer]
If you're talking about ciphers used in the TLS protocol, then in case TLS-1.2 version is negotiated and the peer supports AES-GCM ciphers they will be selected by default. You can also use `AESGCM:-aNULL` as the ciphersuite string to limit TLS to use only AES-GCM for encryption.

**Reply by sireeshkumartholeti** [reply]
Hello,
Thanks for the response,
We are limiting TLS to use only AES-GCM for encryption. However there is no communication/handshake happening between client and server.
We need assistance in properly configuring AES-GCM ciphers as the default without encountering any errors. Additionally, we would like to know which APIs are responsible for handling this configuration.

**Reply by t8m** [reply]
Is both client and server using OpenSSL 1.0.2?

**Reply by sireeshkumartholeti** [reply]
Yes, both and client are using the same versions of OpenSSL 1.0.2u

**Reply by t8m** [reply]
And you are using `AESGCM:-aNULL` as the cipher string? Does the server have a certificate properly set? What were you using as cipher string before?

**Reply by sireeshkumartholeti** [reply]
We are limiting TLS to use only these GCM ciphers.(TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_DHE_RSA_WITH_AES_256_GCM_SHA384,TLS_DHE_RSA_WITH_AES_128_GCM_SHA256) 
since then we are facing issue.

In the process we are creating KeyStore and TrustStore and we are using those for the server. Do we have any specific certificate related to ciphers that we need to include in our KeyStore and TrustStore.
